Modules/Refiner/RefinerFactory.py

```diff
@@ -27,14 +29,24 @@ def RefinerForXSweep(xsweep, json_file=None):
     if not xsweep.__class__.__name__ == "XSweep":
         raise RuntimeError("XSweep instance needed")
 
-    refiner = Refiner()
+    refiner = xsweep.sample.multi_refiner or Refiner()
+
+    multi_sweep_refinement = PhilIndex.params.xia2.settings.multi_sweep_refinement
+    if multi_sweep_refinement and len(xsweep.sample.get_sweeps()) > 1:
+        xsweep.sample.multi_refiner = refiner
 
     if json_file is not None:
         assert os.path.isfile(json_file)
         logger.debug("Loading refiner from json: %s", json_file)
         refiner = refiner.__class__.from_json(filename=json_file)
 
-    refiner.add_refiner_sweep(xsweep)
+    if xsweep.sample.multi_refiner:
+        if not refiner._refinr_sweeps:
+            refiner._refinr_sweeps = xsweep.sample.get_sweeps()
+            for sweep in refiner._refinr_sweeps:
+                sweep._get_refiner()
```

Comment:
OK.

Reply:
Hmm... Perhaps it's not necessary.  I think it may have crept in at a stage where I was getting very confused about when/why refiners were being created and reset.  It basically copies the logic at https://github.com/xia2/xia2/blob/ca2f126a668e2d0ffc7e5018f43b7a83b91c89e8/Modules/Indexer/IndexerFactory.py#L112-L115 Perhaps if this isn't necessary, nor is that?

Removing the line you refer to in `RefinerFactory` doesn't _seem_ to break anything...

Reply:
Ah, yes it does.

Reply:
I know this is a seemingly simple request but it is costing me blood sweat and tears.  I'm going to have to ignore it, sorry.